### DISCUSSION TITLE
Component callbacks in Systems

### DISCUSSION THREAD

**Discussion by Epsola** [SanderMertens/flecs#511] [Q&A]
Hello,

I have an Activity component which has a callback to perform when the activity is considered "finished". 

e.g (code from within the activity system function)

	...
	if (anim->state == AnimationState::Finished)
	{
		act->state = ActivityState::Finished;
		act->activityTime = .0f;

		act->callback();
	}

However, if the callback calls a function that sets an entity's component values (e.g `entity.set<ActivityComponent>({..})`), it results in an assert error. 

Is something like this possible? Or how would I call a callback from within a system without this issue?

Using v2.4.7

**Edit:** 
I've since found that as long as I pass the pointer to the current iter entity to the callback, the set functions will now work. However if I try to get the entity by id instead from within the callback function, it doesn't work. 

I'll leave this here in case anyone wants to add anything new.

Thank you.

**Comment by SanderMertens** [comment]
@Epsola could you share the code that reproduces this issue and the assert that is triggered?
